src/linear_algebra/multilinear.lean

```diff
@@ -437,15 +461,83 @@ instance : has_scalar R (multilinear_map R M₁ M₂) := ⟨λ c f,
 
 @[simp] lemma smul_apply (c : R) (m : Πi, M₁ i) : (c • f) m = c • f m := rfl
 
+section
+
+variables (R ι) (A : Type*) [comm_semiring A] [algebra R A] [fintype ι]
+
+/-- Given an `R`-algebra `A`, `mk_pi_algebra` is the multilinear map on `A^ι` associating
+to `m` the product of all the `m i`.
+
+See also `multilinear_map.mk_pi_algebra_fin` for a version that works with a non-commutative
+algebra `A` but requires `ι = fin n`. -/
```

Comment:
Should `ι` be implicit here to match `multilinear_map`?

Reply:
(sorry, intended for the line below)

Reply:
Given `mk_pi_algebra R A`, how should Lean figure out `ι`? Of course, once it has `m : ι → A`, it should be able to do this but the rest of the code in this file chooses implicit/explicit so that the type is known before we have `m`.